### PR TITLE
Fix travis tests for Rails 5.1+

### PR DESCRIPTION
**Pull request by NickLaMuro**
Don't use `before_filter` for test rails_app since Rails changed `before_filter` to become `before_action`, so prefer using that over before_filter (since it has been removed in the [newest versions of Rails](https://github.com/rails/rails/blob/158742e/actionpack/CHANGELOG.md).

> Remove deprecated methods `skip_action_callback`, `skip_filter`, `before_filter`, `prepend_before_filter`, `skip_before_filter`, `append_before_filter`, `around_filter` `prepend_around_filter`, `skip_around_filter`, `append_around_filter`, `after_filter`, `prepend_after_filter`, `skip_after_filter` and `append_after_filter`.
> 
> Rafael Mendonça França

Also, since we are making `before_action` backwards compatible for older versions of Rails where `before_filter` was the only method available, move that alias builder to the `ApplicationController` instead of the `AuthenticatedController` which was where it lived prior.  This might not make to much sense to do since we aren't using something like [appraisal](https://github.com/thoughtbot/appraisal) in our test suite, but left it in for now.

**Update**:  A fix to the migrations was also required for this to work properly.  See the [second commit](https://github.com/schneems/derailed_benchmarks/pull/105/commits/95939192543bf2c19d1db04a014e9c621205fe81) for details.

### CODE REVIEW

**Comment by NickLaMuro** [comment]
Update:  Looking into fixing the migrations as well.

**Comment by schneems** [comment]
You rock! Is this ready to go?

**Comment by NickLaMuro** [comment]
@schneems yeah, should be.  I am working on getting [appraisal](https://github.com/thoughtbot/appraisal) up and running (if you think that is something you want), but I figured that is better to be done in a separate PR anyway, and can be evaluated from there.

tl; dr;  :shipit:

**Comment by schneems** [comment]
thanks for this!
